### PR TITLE
Add sidebar tutorial for chat models

### PR DESCRIPTION
**Pull request by Rocketknight1**
~Draft for now~ - this PR will add a sidebar tutorial on chat models!

### CODE REVIEW

**Comment by Rocketknight1** [comment]
This should be ready for review! cc @stevhliu. Also cc @zucchini-nlp and @gante since I'm talking about generation-related topics a lot, let me know if there's anything I'm missing, or if I should be tying this into the existing generate docs better!

File: docs/source/en/conversations.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+### Specialist domains
+Some models may be specialized for certain domains, such as medical or legal text, or non-English languages. 
+If you're working in these domains, you may find that a specialized model will give you big performance benefits. 
+Don't automatically assume that, though! Particularly when specialized models are smaller or older than the current 
+cutting-edge, a top-end general-purpose model may still outclass them. Thankfully, we are beginning to see 
+[domain-specific leaderboards](https://huggingface.co/blog/leaderboard-medicalllm) that should make it easier to locate
+the best models for specialized domains.
+
+## Breaking it down: What happens inside the pipeline?
```
Comment (stevhliu) [inline docs/source/en/conversations.md:159]:
```suggestion
## What happens inside the pipeline?
```

File: docs/source/en/conversations.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+The quickstart guide above used a high-level pipeline to chat with a chat model, which is convenient, but not the
```
Comment (stevhliu) [inline docs/source/en/conversations.md:161]:
```suggestion
The quickstart above used a high-level pipeline to chat with a chat model, which is convenient, but not the
```

File: docs/source/en/conversations.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+There's a lot in here, each piece of which could be its own document! Rather than going into too much detail, I'll cover
+the broad ideas, and leave the details for the linked documents. The key steps are:
+
+1. [Models](https://huggingface.co/learn/nlp-course/en/chapter2/3) and [Tokenizers](https://huggingface.co/learn/nlp-course/en/chapter2/4?fw=pt) are loaded from the Hugging Face Hub.
+2. The chat is formatted using the tokenizer's [chat template](https://huggingface.co/docs/transformers/main/en/chat_templating)
+3. The formatted chat is [tokenized](https://huggingface.co/learn/nlp-course/en/chapter2/4) using the tokenizer.
+4. We [generate](https://huggingface.co/docs/transformers/en/llm_tutorial) a response from the model.
+5. The tokens output by the model are decoded back to a string
+
+## Advanced: Performance, memory and hardware
```
Comment (stevhliu) [inline docs/source/en/conversations.md:207]:
```suggestion
## Performance, memory and hardware
```

File: docs/source/en/conversations.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+Or we can do the same thing using the `pipeline` API:
+
+```python
+from transformers import pipeline, BitsAndBytesConfig
+
+quantization_config = BitsAndBytesConfig(load_in_8bit=True)  # You can also try load_in_4bit
+pipe = pipeline("text-generation", "meta-llama/Meta-Llama-3-8B-Instruct", device_map="auto", model_kwargs={"quantization_config": quantization_config})
+```
+
+There are several other options for quantizing models besides `bitsandbytes` - please see the [Quantization guide](https://huggingface.co/docs/transformers/en/quantizationhttps://huggingface.co/docs/transformers/en/quantization)
```
Comment (stevhliu) [inline docs/source/en/conversations.md:242]:
```suggestion
There are several other options for quantizing models besides `bitsandbytes` - please see the [Quantization guide](./quantization)
```

File: docs/source/en/conversations.md [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+for more information.
+
+### Performance considerations
```
Comment (stevhliu) [inline docs/source/en/conversations.md:245]: I think it'd also be helpful to link to the [LLM inference optimization guide](https://moon-ci-docs.huggingface.co/docs/transformers/pr_30401/en/llm_optims) which has code examples for assisted generation, quantization, and attention optimizations.

File: docs/source/en/_toctree.yml [hunk 6]
```diff
@@ -27,6 +27,8 @@
     title: Agents
   - local: llm_tutorial
     title: Generation with LLMs
+  - local: conversations
+    title: Talking to Chat Models
```
Comment (stevhliu) [inline docs/source/en/_toctree.yml:31]:
```suggestion
    title: Chatting with Transformers
```

File: docs/source/en/conversations.md [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# Chatting with Transformers: A Beginner's Guide
```
Comment (stevhliu) [inline docs/source/en/conversations.md:17]:
```suggestion
# Chatting with Transformers
```

**Review by stevhliu — APPROVED** [review]
Very nice and thorough tutorial! There is a lot of info here but I didn't feel like it was too much detail. I think it does a great job of the mental walkthrough and considerations (choosing a model, performance, etc.) you need to make before you even touch any of the code. 👏

File: docs/source/en/conversations.md [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+    {"role": "system", "content": "You are a sassy, wise-cracking robot as imagined by Hollywood circa 1986."},
+    {"role": "user", "content": "Hey, can you tell me any fun things to do in New York?"}
+]
+```
+
+Notice that in addition to the user's message, we added a **system** message at the start of the conversation. Not all
+chat models support system messages, but when they do, they represent high-level directives about how the model
+should behave in the conversation.
+
+Once you have a chat, the quickest way to continue it is using the [Text Generation pipeline](https://huggingface.co/meta-llama/Meta-Llama-3-8B-Instruct).
```
[the remark this answers is not in the document]
Comment (Rocketknight1) [reply docs/source/en/conversations.md:50]: Yes, this is a mistake. Will fix!

File: docs/source/en/conversations.md (hunk 5 above)
Comment (Rocketknight1) [reply docs/source/en/conversations.md:249]: Good catch! I'll add that as well.

**Comment by gante** [comment]
@Rocketknight1 cool tutorial 🙌

**Pull request merged**